### PR TITLE
`riak-cs-debug` update

### PR DESCRIPTION
**Pull request by pyrrho**
There were a lot of updates made to `riak-debug` that have not yet been pulled into `riak-cs-debug`. This PR looks to correct that.

These changes have not yet been tested, though, and definitely need to be checked on Ubuntu and Solaris before being merged.

### CODE REVIEW

**Comment by kuenishi** [comment]
Looks nice, but I don't have Solaris env. Anybody?

**Comment by slfritchie** [comment]
The script will hang forever if there is no `rel/riak-cs/bin/.././data/generated.configs/app.*.config` file, e.g., because CS hasn't been run for the first time yet.

But aside from that, it appears to work correctly on Solaris: `SunOS oi151a-64-0 5.11 oi_151a i86pc i386 i86pc Solaris`

File: rel/files/riak-cs-debug [hunk 1]
```diff
@@ -325,6 +372,9 @@ if [ 1 -eq $get_syscmds ]; then
     dump netstat_rn netstat -rn
     dump pfctl_rules pfctl -s rules
     dump pfctl_nat pfctl -s nat
+    dump java_version java -version
```
Comment (ksauzz) [inline rel/files/riak-cs-debug:375]: Is this necessary?

**Comment by ksauzz** [comment]
On Ubuntu, this script failed:

```
root@vagrant-ubuntu-trusty-64:~# riak-cs-debug
....EE../usr/sbin/riak-cs-debug: 577: [: !: unexpected operator
. /root/riak-cs@127.0.0.1-riak-cs-debug.tar.gz
```

File: rel/files/riak-cs-debug [hunk 2]
```diff
@@ -487,23 +555,44 @@ fi
 
 if [ 1 -eq $get_cfgs ]; then
     mkdir_or_die "${start_dir}"/"${debug_dir}"/config/.info
-    cd "${start_dir}"/"${debug_dir}"/config
 
-    # Use dump to execute the copy. This will provide a progress dot and
-    # capture any error messages.
-    dump cp_cfg cp -R "$cs_etc_dir"/*.conf* .
```
Comment (ksauzz) [inline rel/files/riak-cs-debug:494]: Gathering riak-cs.conf and advanced.config is also helpful for CSEs to figure out a mistake on the configs, isn't it?

File: rel/files/riak-cs-debug [hunk 3]
```diff
@@ -487,23 +555,44 @@ fi
...
+        for i in `echo "'*.pfx' '*.p12' '*.csr' '*.sst' '*.sto' '*.stl' '*.pem' '*.key'"`; do
+            exclude="${exclude}! -name $i "
+        done
     fi
 
-    mkdir_or_die "${start_dir}"/"${debug_dir}"/generated.config/.info
-    cd "${start_dir}"/"${debug_dir}"/generated.config
+    # Compose the `find` command that will exclude the above list of files,
+    # being aware that an empty `\( \)` will cause a failure in the `find`.
+    if [ -n $exclude ]; then
```
Comment (ksauzz) [inline rel/files/riak-cs-debug:577]: It seems `$exclude` needs double-quote for avoiding error when $exclude is empty.

File: rel/files/riak-cs-debug [hunk 4]
```diff
@@ -455,23 +518,28 @@ if [ 1 -eq $get_logs ]; then
...
 
             lager_file="`echo $lager_path | awk -F/ '{print $NF}'`"
-            lager_dir="`echo $lager_path | awk -F/$lager_file '{print $1}'|sed 's|/./|/|'`"
+            lager_dir="`echo $lager_path | awk -F/$lager_file '{print $1}'`"
             if [ "$cs_log_dir" != "$lager_dir" ]; then
-                if [ -n "`find "${lager_dir}" -maxdepth 1 -mtime -$log_mtime -name "${lager_file}*" -print -quit`" ]; then
-                    mkdir_or_die "${start_dir}"/"${debug_dir}"/logs/lager_"${lager_num}"
+                ls -1 "${lager_dir}" | grep -q "${lager_file}"
+                if [ 0 -eq $? ]; then
+                    mkdir_or_die "${start_dir}"/"${debug_dir}"/logs/lager_${lager_level}
```
Comment (ksauzz) [inline rel/files/riak-cs-debug:542]: Although I'm not sure what's wrong, this part looks not to work. Lager's logs are not gathered in the case which console.log and error.log are out of platform_log_dir such as:

```
platform_log_dir = /var/log/riak-cs
log.console.file = /data/log/console.log
log.error.file = /data/log/error.log
```

**Comment by ksauzz** [comment]
Thank you for the effort to update this from latest riak-debug. :bowtie: I left a few comments.

**Comment by shino** [comment]
Changes are included in #1236 (RCS-284).

**Pull request closed without merging**